.github/workflows/master.yml

```diff
@@ -71,6 +71,11 @@ jobs:
             spark-archive: '-Dspark.archive.mirror=https://archive.apache.org/dist/spark/spark-3.4.2 -Dspark.archive.name=spark-3.4.2-bin-hadoop3.tgz -Pzookeeper-3.6'
             exclude-tags: '-Dmaven.plugin.scalatest.exclude.tags=org.scalatest.tags.Slow,org.apache.kyuubi.tags.DeltaTest,org.apache.kyuubi.tags.IcebergTest,org.apache.kyuubi.tags.PaimonTest,org.apache.kyuubi.tags.SparkLocalClusterTest'
             comment: 'verify-on-spark-3.4-binary'
+          - java: 21
+            spark: '3.5'
```

Comment:
FYI. 03dcedd89ed26fb21d0475b974a637e805e06d3b enabled CI for Spark 4.0, with JDK 17, and Spark 4.0 is fully compatible with Java 21.